### PR TITLE
#14 created page template

### LINKED ISSUE #14: Code a template for all the pages the app will need

### PR DESCRIPTION
**Pull request by Tibesnoff**
Created page templates for component, api, and both

### CODE REVIEW

File: app/tools/generate.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
+/* eslint-disable quotes */
```
Comment (alexanderjhughes) [inline app/tools/generate.js]: Lets not disable quotes, just keeps everything consistent

File: app/tools/templates/componentAndAPI/component/__name__.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import type {
+  /*__screen__NavigationRoutesType,*/
+  NavigationRoutesType,
+} from '../../..__src__(camelCase)/types/NavigationRoutesType';
+import type {NativeStackScreenProps} from '@react-navigation/native-stack';
+import type {BottomTabScreenProps} from '@react-navigation/bottom-tabs';
+import type {CompositeScreenProps} from '@react-navigation/native';
+
+type __screen__ScreenProps = /*CompositeScreenProps<
+NativeStackScreenProps<__screen__NavigationRoutesType, '__screen__Screen'>,*/
```
Comment (alexanderjhughes) [inline app/tools/templates/componentAndAPI/component/__name__.tsx]: add tab in front to fix eslint error on generating new file

Comment (alexanderjhughes) [reply app/tools/templates/componentAndAPI/component/__name__.tsx]: Would look like this
```
type __screen__ScreenProps = /*CompositeScreenProps<
  NativeStackScreenProps<__screen__NavigationRoutesType, '__screen__Screen'>,*/
  BottomTabScreenProps<NavigationRoutesType>
/*>*/ &
  PropsWithChildren;
```

**Review by alexanderjhughes — CHANGES_REQUESTED** [review]
When generating a component with an api, couple quick questions.

1. Can we generate the components in the screen folder instead of generating a new sub folder named component? So instead of Screens -> Location -> components -> newComponent.tsx, instead we just put it in the Location directory. 
2. What do you mean by API? The API is the backend, I am just confused why we are calling it API on front end. Its just loading information from the server. 
3. Why segregate the fetch in a separate file? Why not just add it in the component file as a function. Imo its a little too much organizing and adds complexity. Maybe its good to have it this way idk. 

One other thing, creating just an api makes a directory called __name__(pascalcase) then puts the api in that directory.

**Comment by Tibesnoff** [comment]
> When generating a component with an api, couple quick questions.
> 
> 1. Can we generate the components in the screen folder instead of generating a new sub folder named component? So instead of Screens -> Location -> components -> newComponent.tsx, instead we just put it in the Location directory.
> 2. What do you mean by API? The API is the backend, I am just confused why we are calling it API on front end. Its just loading information from the server.
> 3. Why segregate the fetch in a separate file? Why not just add it in the component file as a function. Imo its a little too much organizing and adds complexity. Maybe its good to have it this way idk.
> 
> One other thing, creating just an api makes a directory called **name**(pascalcase) then puts the api in that directory.



1. I think it has something to do with how it generates from a folder I will loook at it.
2. When I mean api I think I mean more just grabbing the backend. I personally think it may be better just to have these fetches seperate so component code doesn't end up being too long.

**Comment by Tibesnoff** [comment]
> When generating a component with an api, couple quick questions.
> 
> 1. Can we generate the components in the screen folder instead of generating a new sub folder named component? So instead of Screens -> Location -> components -> newComponent.tsx, instead we just put it in the Location directory.
> 2. What do you mean by API? The API is the backend, I am just confused why we are calling it API on front end. Its just loading information from the server.
> 3. Why segregate the fetch in a separate file? Why not just add it in the component file as a function. Imo its a little too much organizing and adds complexity. Maybe its good to have it this way idk.
> 
> One other thing, creating just an api makes a directory called **name**(pascalcase) then puts the api in that directory.

I changed it up a lot and removed the api stuff as well as added tailwinds

File: app/src/Components/Screens/Search/SearchText.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import {styled} from 'nativewind';
+import {Text} from 'react-native';
+import ComponentPropType from '../../../types/ComponentPropType';
+
+const StylizedTouch = styled(Text);
```
Comment (alexanderjhughes) [inline app/src/Components/Screens/Search/SearchText.tsx]: Use a TouchableHighlight, not Text: [Link to Docs](https://reactnative.dev/docs/handling-touches)

File: app/tools/templates/component/__name__.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import {styled} from 'nativewind';
+import {Text} from 'react-native';
+import ComponentPropType from '../../../types/ComponentPropType';
+//All of this is meant to be replaced it just shows the basic structure of the component
+//Navigation does not need to be included if it is not needed
+const StylizedTouch = styled(Text);
```
Comment (alexanderjhughes) [inline app/tools/templates/component/__name__.tsx]: TouchableHighlight or a different touchable

File: app/tools/generate.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+      folderPath: './tools/templates/screen',
+    },
+    stringReplacers: [
+      {
+        question: 'What is the name of this screen?',
+        slot: '__screen__',
+      },
+      {
+        question: 'What is the name of the folder for this screen?',
+        slot: '__folder__',
```
Comment (alexanderjhughes) [inline app/tools/generate.js]: __stack__

**Review by alexanderjhughes — CHANGES_REQUESTED** [review]

**Pull request merged**